editor/code_editor.cpp

```diff
@@ -1473,6 +1489,12 @@ void CodeTextEditor::_on_settings_change() {
 
 	font_size = EditorSettings::get_singleton()->get("interface/editor/code_font_size");
 
+	if (bool(EditorSettings::get_singleton()->get("interface/editor/code_font_ligatures"))) {
```

Comment:
That is a very specific name for a general feature. `calt` might not be used for ligatures (an Arabic or Indic monospaced font for instance), and ”coding ligatures” might use other features like `liga`. Either make the name reflect what the code is actually doing (not the its intent) or make it a general OpenType features control.

Reply:
> or make it a general OpenType features control.

The Editor Settings dialog can only expose simple booleans, integers, floats or "fixed" enums, not an enum that's dynamically constructed based on the currently loaded font.

Reply:
A fixed `enum` should work, the [number of OpenType features](https://docs.microsoft.com/en-us/typography/opentype/spec/featurelist) is finite, and though the list is long, many of the features are required (for proper text layout) and should not be controllable.

Reply:
Fixed enum won't give full control, since some features can be set to the integer value not just enabled/disabled (swashes and other stylistic features), and each feature can be applied to the sub range of the string.

For generic controls, OT features are currently set as the single string property (e.g `swsh=2,lnum,zero`), but for the code editor it's probably better to have some fixed list too, maybe:

- `code_font_ligatures` → (enum) "Default, Disable ligatures, Use custom OT features"
- `code_font_ot_features` → (string) for full control, used if "custom" is selected.

Reply:
> `code_font_ligatures` → (enum) "Default, Disable ligatures, Use custom OT features"

My point is that a name like `code_font_ligatures` is misleading, if you really want to control `calt` (though I’m pretty sure there are ”coding fonts” using `liga`), just call it what it is; contextual ligatures.

Reply:
At least "Fira Code", "Iosevka" and "JetBrains Mono" use `calt` to control coding ligatures `liga` do not affect it at all.

Reply:
I guess calling it `code_font_contextual_ligatures` instead of `code_font_ligatures` won't hurt, it's still easy to find by searching for "ligatures", but on the other hand `calt` is "contextual alternates" not "contextual ligatures" (`clig` which do not have any effect on coding ligatures).

Reply:
Additionally, some coding fonts seem to have non-standard features for specific subsets of ligatures, like `HSKL` for Haskell ligatures and `CLIK` for C ligatures.

Reply:
Changed it to `Default, Disable coding ligatures, Use custom OpenType feature set` enum + custom features string.

<img width="864" alt="Screenshot 2020-10-18 at 22 19 15" src="https://user-images.githubusercontent.com/7645683/96377726-32318100-1190-11eb-9cde-cf9520ae9694.png">

Reply:
Great, thanks. (`Ot` should probably be `OT` or better `OpenType`).

Reply:
> Ot should probably be OT or better OpenType

It should be, but currently editor setting names are automatically converted from snake case to title case, AFAIK there's no way to archive this (there are many other setting that should be all upper case abbreviations but are not for the same reason).

Reply:
Naming it `Opentype` would be better. (It's all we can do with the current automatic capitalization; we'd need https://github.com/godotengine/godot/pull/32734 to be merged to improve this.)

Reply:
Renamed `ot` to `opentype` (editor setting and generic control properties), and added simple editor inspector plugin to edit features:
<img width="467" alt="Screenshot 2020-10-19 at 12 13 18" src="https://user-images.githubusercontent.com/7645683/96425567-d3f2b580-1204-11eb-9f7e-7c9c221f7f85.png">